sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -100,7 +100,7 @@ abstract class JdbcDialect extends Serializable {
    * @return The SQL query to use for checking the table.
    */
   def getTableExistsQuery(table: String): String = {
-    s"SELECT * FROM $table WHERE 1=0"
+    s"SELECT * FROM ${quoteIdentifier(table)} WHERE 1=0"
```

Comment:
This is a behavior change. After we quote the names, the table names become case sensitive in most RDBMS. We need at least a JDBC option to control it.